src/index.ts

```diff
@@ -0,0 +1,248 @@
+import type { AxiosError, AxiosRequestConfig, AxiosInstance, AxiosStatic } from 'axios';
+import isRetryAllowed from 'is-retry-allowed';
+
+export namespace IAxiosRetry {
```

Comment:
@mindhells 

If you are moving to typescript, I don't think you need to bother with namespace. Also, the presence of this namespace would be a destructive change.

Specifically, this code does not export `IAxiosRetryConfig`, etc.

I would appreciate it if you could refer to the PR of the proposed fix below for more details.

https://github.com/softonic/axios-retry/pull/256

Reply:
If we want to keep namespace, you would need to export each one separately as follows. However, this is redundant, so I think a modification that eliminates the namespace declaration is fine.

```ts
export type AxiosRetry= IAxiosRetry.AxiosRetry;
export type IAxiosRetryConfig = IAxiosRetry.IAxiosRetryConfig;
export type IAxiosRetryConfigExtended = IAxiosRetry.IAxiosRetryConfigExtended;
export type IAxiosRetryReturn = IAxiosRetry.IAxiosRetryReturn;
```